cf-agent/verify_files_utils.c

```diff
@@ -3243,14 +3244,22 @@ static void SaveSetxid(bool modified)
     {
         char filename[CF_BUFSIZE];
         snprintf(filename, CF_BUFSIZE, "%s/cfagent.%s.log", GetLogDir(), VSYSNAME.nodename);
+        ToLowerStrInplace(filename);
         MapName(filename);
 
         PurgeItemList(&VSETXIDLIST, "SETUID/SETGID");
 
         Item *current = RawLoadItemList(filename);
         if (!ListsCompare(VSETXIDLIST, current))
         {
+            mode_t oldmode = umask(077); // This setxidlist file must only be accesible by root
```

Comment:
Why did you change it to 077?

Reply:
I was looking at the bug report and it asked for 0600 permissions.

Reply:
@olehermanse @nickanderson Does this look OK to you? umask set to 077 to achieve 0600 in the end.

Reply:
@vpodzime @craigcomstock As I understand it umask will mask permissions using bitwise AND, so:
```
const int restore = umask(0700);
FILE *f = fopen("file.txt", "w");
umask(restore);
```

Will create a file with permissions `0600`, since `0666` is the default, and the zeroes in the mask will cancel out the 2 last sixes.

Reply:
```
Specifically, permissions in the umask are turned off from the mode argument to open(2) and mkdir(2).
```
(from man:umask(2))

Reply:
@olehermanse as @vpodzime mentioned, from reading umask(2) and open(2) I see resulting permissions being `mode & ~umask` so zeroes in a mask will allow permissions through and sevens in a mask will block them out.

On my debian I have umask as 0022 (from command line command) and when I create a file with fopen() I get 644 as a result. If I do umask(0700) I get 0066 which is not what I want. I think 077 is right as it will always block out any permissions for group and other and cause the system call's mode parameter to affect only the user permissions.

Reply:
I agree. Based on my research it should be 077 umask to ensure that files are read/write by owner, directories are read/write/execute by owner, and nothing else.

```
    umask    077
    File Permissions    rw-------
    Directory Permissions    rwx------
```
http://www.webune.com/forums/umask-calculator.html